lib/ansible/modules/cloud/vmware/vmware_local_user_manager.py

```diff
@@ -111,71 +132,240 @@ def process_state(self):
+                results['msg'] = "User account would be updated"
+            else:
+                account_spec = self.create_account_spec()
+                try:
+                    self.content.accountManager.UpdateUser(account_spec)
+                except vim.fault.UserNotFound as user_not_found:
+                    self.module.fail_json(msg="The local user account wasn't found : %s" % to_native(user_not_found))
+                except vim.fault.AlreadyExists as already_exists:
+                    self.module.fail_json(msg="The specified local user account already exists : %s" % to_native(already_exists))
+                except vmodl.fault.InvalidArgument as invalid_argument:
+                    self.module.fail_json(
+                        msg="The new password or description has an invalid format : %s" % to_native(invalid_argument)
+                    )
+                if self.local_user_role:
+                    role_id = self.get_role_id(self.local_user_role, self.content.authorizationManager.roleList)
+                    self.set_permissions(self.local_user_name, role_id)
+                results['msg'] = "User account updated"
+        else:
+            results['msg'] = "User account is properly configured"
+        self.module.exit_json(changed=changed, result=results)
 
     def state_remove_user(self):
-        try:
-            self.content.accountManager.RemoveUser(self.local_user_name)
-            self.module.exit_json(changed=True)
-        except vmodl.RuntimeFault as runtime_fault:
-            self.module.fail_json(msg=runtime_fault.msg)
-        except vmodl.MethodFault as method_fault:
-            self.module.fail_json(msg=method_fault.msg)
+        """Remove the user account"""
+        results = dict()
+        results['user_name'] = self.local_user_name
+        results['description'] = self.local_user_description
+        results['role'] = self.local_user_role
+        if self.module.check_mode:
+            results['msg'] = "User account would be removed"
+        else:
+            try:
+                self.content.accountManager.RemoveUser(self.local_user_name)
+                results['msg'] = "User account removed"
+            except vim.fault.UserNotFound as user_not_found:
+                self.module.fail_json(msg="The specified user account doesn't exist : %s" % to_native(user_not_found))
+            except vmodl.fault.SecurityError as security_error:
+                self.module.fail_json(
+                    msg="Failed to remove the user account, this can be due to either of following : "
+                    "1. Trying to remove the last local user with DCUI access, "
+                    "2. Trying to remove the last local user with full administrative privileges, "
+                    "3. The system has encountered an error while trying to remove user's permissions, "
+                    "4. The account cannot be removed due to permission issues : %s" % to_native(security_error)
```

Comment:
```suggestion
                    "4. The account cannot be removed due to permission issues : %s" % to_native(security_error.msg)
```

Reply:
If you don't specify msg, then the whole object will be in the error output. It's not that clean compared to only msg, but then you have more information available than in the GUI. For some errors the msg property don't even exist.

Reply:
ok.